dissect/cstruct/utils.py

```diff
@@ -173,6 +168,10 @@ def _dumpstruct(
                 value = value.replace("\n", f"\n{' ' * (len(field._name) + 4)}")
 
         if color:
+            foreground, background = colors[ci % len(colors)]
+            size = structure._sizes[field._name] if field.type.dynamic else field.type.size
```

Comment:
Reviewer: This logic could be moved into a `sizes` property on the cstruct type.